### PR TITLE
Window contribute support

### PR DESCRIPTION
**Pull request by KingDarBoja**
### Checklist

   - Issue number for this PR: #2531 (required)
   - Docs included?: no
   - Test units included?: no
   - In a clean directory, `yarn install`, `yarn test` run successfully? yes

### Description

Pretty much tried to run the build on my Windows 10 machine but got several errors explained on linked issue.
This PR aims to fix those compatibility issues and also provide GitHub Actions matrix option to run the tests on different OS.

Not sure about the tests itself as had to change a little bit the path separator to make it work 🚀

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -37,6 +37,9 @@
   "bugs": {
     "url": "https://github.com/angular/angularfire/issues"
   },
+  "engines": {
+    "node": "^8.13.0 || >=10.10.0"
```
Comment (yasinkocak) [inline package.json:41]: node lts version is 12, node 8 is very behind

File: src/schematics/ng-add.jasmine.ts [hunk 2]
```diff
@@ -368,7 +369,7 @@ describe('ng-add', () => {
     });
 
     it('generates new files if starting from scratch', async () => {
-      const result = await setupProject(tree, {
```
Comment (yasinkocak) [inline src/schematics/ng-add.jasmine.ts:371]: if `await` not needed then you also have to remove `async`, also applies this to the rest where `await` is removed

**Review by yasinkocak — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (KingDarBoja) [reply package.json:41]: I used the same engine requirement as firebase-js sdk: https://github.com/firebase/firebase-js-sdk/blob/master/package.json#L9

**Comment by jamesdaniels** [comment]
Updated and merged in https://github.com/angular/angularfire/commit/27889806b87f5bcff028561d392b4d80824fdd1b, thanks for getting the ball rolling here

**Pull request closed without merging**